### DISCUSSION TITLE
Batched upsert

### DISCUSSION THREAD

**Discussion by dullbananas** [diesel-rs/diesel#3878] [Ideas]
This could be done by adding a `set_excluded` function that takes an `AsChangeset` implementation and applies `excluded` to the columns that don't have the corresponding field set to `None`

```
diesel::insert_into(bunny::table)
    .values(&new_bunnies)
    .on_conflict(id)
    .do_update()
    .set(set_excluded(&new_bunnies))
```

Edit: actually this idea doesn't work because the items in `new_bunnies` could have varying `Some` or `None` values, but another thing that could be done which is not exactly the same is `set_excluded(bunny::all_columns)`

**Comment by weiznich** [comment]
For Feature Request proposals we expect quite a lot more details than: "I want to have this or that".

I would expect at least the following information:

- How does the exact proposed API look like
- What invariants need to be uphold
- What database systems support this syntax
- How would this be implemented (You don't need to provide a complete implementation, but you should at least cover some sketch how something can be represented in the type system)
